internal/server/ingest/ingest.go

```diff
@@ -0,0 +1,229 @@
+// Package ingest is responsible for running the ingest-service in the background.
+package ingest
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"log/slog"
+	"path/filepath"
+	"sync"
+	"time"
+
+	"github.com/ubuntu/ubuntu-insights/internal/server/ingest/database"
+	"github.com/ubuntu/ubuntu-insights/internal/server/ingest/models"
+	"github.com/ubuntu/ubuntu-insights/internal/server/ingest/processor"
+)
+
+// Service represents the ingest service that processes and uploads reports to the database.
+type Service struct {
+	cm dConfigManager
+	db dbManager
+
+	// This context is used to interrupt any action.
+	// It must be the parent of gracefulCtx.
+	ctx    context.Context
+	cancel context.CancelFunc
+
+	// This context waits until the next blocking Recv to interrupt.
+	gracefulCtx    context.Context
+	gracefulCancel context.CancelFunc
+
+	mu      sync.Mutex
+	workers map[string]context.CancelFunc
+}
+
+type dbManager interface {
+	Upload(ctx context.Context, app string, data *models.TargetModel) error
+	Close() error
+}
+
+type dConfigManager interface {
+	Load() error
+	Watch(context.Context) (<-chan struct{}, <-chan error, error)
+	AllowList() []string
+	BaseDir() string
+}
+
+type options struct {
+	dbConnect func(ctx context.Context, cfg database.Config) (dbManager, error)
+}
+
+// Options is a function that modifies the options for the ingest service.
+type Options func(*options)
+
+// New creates a new ingest service with the provided database manager and connects to the database.
+func New(ctx context.Context, cm dConfigManager, dbConfig database.Config, args ...Options) (*Service, error) {
+	opts := options{
+		dbConnect: func(ctx context.Context, cfg database.Config) (dbManager, error) {
+			return database.Connect(ctx, cfg)
+		},
+	}
+
+	for _, opt := range args {
+		opt(&opts)
+	}
+
+	if err := cm.Load(); err != nil {
+		return nil, fmt.Errorf("failed to load configuration: %v", err)
+	}
+
+	dctx, dcancel := context.WithTimeout(ctx, 10)
```

Comment:
```suggestion
	dctx, dcancel := context.WithTimeout(ctx, 10 * time.Second)
```

I'd be surprised if connecting to the database would reliably happen under 10 nanoseconds. :)